dev/devicelab/lib/tasks/hot_mode_tests.dart

```diff
@@ -26,74 +28,76 @@ TaskFunction createHotModeTest({String deviceIdOverride, Map<String, String> env
     rm(benchmarkFile);
     final List<String> options = <String>[
-      '--hot', '-d', deviceIdOverride, '--benchmark', '--verbose', '--resident',  '--no-android-gradle-daemon',
+      '--hot', '-d', deviceIdOverride, '--benchmark', '--resident',  '--no-android-gradle-daemon',
     ];
     int hotReloadCount = 0;
     Map<String, dynamic> twoReloadsData;
+    Map<String, dynamic> threeReloadsData;
+    Map<String, dynamic> fourReloadsData;
     Map<String, dynamic> freshRestartReloadsData;
+
+
     await inDirectory<void>(flutterDirectory, () async {
       rmTree(_editedFlutterGalleryDir);
       mkdirs(_editedFlutterGalleryDir);
       recursiveCopy(flutterGalleryDir, _editedFlutterGalleryDir);
-      await inDirectory<void>(_editedFlutterGalleryDir, () async {
-        {
-          final Process clearProcess = await startProcess(
-              path.join(flutterDirectory.path, 'bin', 'flutter'),
-              flutterCommandArgs('clean', <String>[]),
-              environment: environment,
-          );
-          await clearProcess.exitCode;
 
-          final Process process = await startProcess(
-              path.join(flutterDirectory.path, 'bin', 'flutter'),
-              flutterCommandArgs('run', options),
-              environment: environment,
-          );
-
-          final Completer<void> stdoutDone = Completer<void>();
-          final Completer<void> stderrDone = Completer<void>();
-          process.stdout
-              .transform<String>(utf8.decoder)
-              .transform<String>(const LineSplitter())
-              .listen((String line) {
-            if (line.contains('] Reloaded ')) {
-              if (hotReloadCount == 0) {
-                // Update the file and reload again.
-                final File appDartSource = file(path.join(
-                    _editedFlutterGalleryDir.path, 'lib/gallery/app.dart',
-                ));
-                appDartSource.writeAsStringSync(
-                    appDartSource.readAsStringSync().replaceFirst(
-                        "'Flutter Gallery'", "'Updated Flutter Gallery'",
-                    )
-                );
-                process.stdin.writeln('r');
-                ++hotReloadCount;
-              } else {
-                // Quit after second hot reload.
-                process.stdin.writeln('q');
-              }
-            }
-            print('stdout: $line');
-          }, onDone: () {
-            stdoutDone.complete();
-          });
-          process.stderr
-              .transform<String>(utf8.decoder)
-              .transform<String>(const LineSplitter())
-              .listen((String line) {
-            print('stderr: $line');
-          }, onDone: () {
-            stderrDone.complete();
-          });
+      await inDirectory<void>(_editedFlutterGalleryDir, () async {
+        twoReloadsData = await captureReloadData(options, environment, benchmarkFile, (String line, Process process) {
```

Comment:
How about `smallReloadData`, `mediumReloadData`, and `largeReloadData` instead of `two`, `three`, `fourReloadsData`?

Reply:
Done